GsaGH/Parameters/0_Model/GsaGridLine.cs

```diff
@@ -184,5 +187,29 @@ internal void UpdatePreview() {
         Circle = new Circle(plane, radius);
       }
     }
+
+    public GridLine GetApiGridLineToUnit(LengthUnit unit) {
+      GridLine gridLine = DuplicateApiObject();
+      gridLine.X = unit == LengthUnit.Meter ? GridLine.X : new Length(GridLine.X, unit).Value;
+      gridLine.Y = unit == LengthUnit.Meter ? GridLine.Y : new Length(GridLine.Y, unit).Value;
+      gridLine.Length = unit == LengthUnit.Meter ? GridLine.Length : new Length(GridLine.Length, unit).Value;
+
+      return gridLine;
+    }
+
+    public GridLine DuplicateApiObject() {
+      var gsaLine = new GridLine(GridLine.Label) {
+        Theta1 = GridLine.Theta1,
+        X = GridLine.X,
+        Y = GridLine.Y,
+        Length = GridLine.Length,
+        Shape = GridLine.Shape,
+      };
+      if (GridLine.Shape == GridLineShape.Arc) {
```

Comment:
Why to check condition when creating duplicate object?
public GridLine DuplicateApiObject() {
  var gridLine = new GridLine(GridLine.Label) {
    Theta1 = GridLine.Theta2,
    Theta2 = GridLine.Theta1,
    X = GridLine.X,
    Y = GridLine.Y,
    Length = GridLine.Length,
    Shape = GridLine.Shape,
  };
  return gridLine;
}

Reply:
because we are not able to set theta2 if shape is arc...